.github/scripts/simple-smoke-test.sh

```diff
@@ -29,6 +29,18 @@ function stop_container() {
     docker stop "${container_name}"
 }
 
+function check_java_version() {
+    local expected_major_version=$1
+    echo "Checking Java version - expected ${expected_major_version}"
+    local actual_major_version
+    actual_major_version=$(docker run --rm "${image}" sh -c 'java -version 2>&1 | head -n 1 | awk -F "\"" "{print \$2}" | awk -F "." "{print \$1}"')
```

Comment:
Against which distribution and java version have you tested the extraction of the major number?

Reply:
> Against which distribution and java version have you tested the extraction of the major number?

Local JDK21 + whatever JDKs are being used when building the various Hazelcast Docker images.

I actually copied this from https://github.com/hazelcast/hazelcast-cpp-client/blob/4a027a85f131d733b44d0ffb32b74b989269e0ec/scripts/start-rc.sh#L121